### PR TITLE
Functional  tests - Add new test 'FO > Consult merchandise returns details'

### PR DESCRIPTION
**Pull request by nesrineabdmouleh**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | 8.0.x
| Description?      | Add new test 'FO > Consult merchandise returns details'
| Type?             | refacto
| Category?         | TE
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | no
| Related PRs       | no
| How to test?      | no need
| Possible impacts? | no

### CODE REVIEW

File: tests/UI/campaigns/functional/FO/03_userAccount/merchandisereturns/01_consultReturnDetails.js [hunk 1]
```diff
@@ -0,0 +1,456 @@
...
+
+        const packageStatus = await foMerchandiseReturnsPage.getTextColumn(page, 'fileName');
+        await expect(packageStatus).to.contains('#RE00');
+      });
+
+      it('should verify the order return status', async function () {
+        await testContext.addContextItem(this, 'testIdentifier', 'checkOrderReturnStatus', baseContext);
+
+        const packageStatus = await foMerchandiseReturnsPage.getTextColumn(page, 'status');
+        await expect(packageStatus).to.equal('Waiting for confirmation');
```
Comment (Progi1984) [inline tests/UI/campaigns/functional/FO/03_userAccount/merchandisereturns/01_consultReturnDetails.js]: Could you move the label to the `foMerchandiseReturnsPage` ?

File: tests/UI/campaigns/functional/FO/03_userAccount/merchandisereturns/01_consultReturnDetails.js [hunk 2]
```diff
@@ -0,0 +1,456 @@
...
+      it('should check the return details', async function () {
+        await testContext.addContextItem(this, 'testIdentifier', 'checkReturnDetails', baseContext);
+
+        const orderReturnInfo = await returnDetailsPage.getOrderReturnInfo(page);
+        await expect(orderReturnInfo)
+          .to.contains('We have logged your return request.')
+          .and.to.contains('Your package must be returned to us within 14 days of receiving your order.')
+          .and.to.contains('The current status of your merchandise return is')
+          .and.to.contains(ReturnStatuses.waitingForConfirmation.name)
+          .and.to.contains('List of items to be returned:')
```
Comment (Progi1984) [inline tests/UI/campaigns/functional/FO/03_userAccount/merchandisereturns/01_consultReturnDetails.js]: Could you move label to `returnDetailsPage` ?

File: tests/UI/campaigns/functional/FO/03_userAccount/merchandisereturns/01_consultReturnDetails.js [hunk 3]
```diff
@@ -0,0 +1,456 @@
...
+          .and.to.contains('Hummingbird printed t-shirt (Size: S - Color: White) Reference: demo_1 1');
```
Comment (Progi1984) [inline tests/UI/campaigns/functional/FO/03_userAccount/merchandisereturns/01_consultReturnDetails.js]: May be could you use demo data ?

File: tests/UI/campaigns/functional/FO/03_userAccount/merchandisereturns/01_consultReturnDetails.js [hunk 4]
```diff
@@ -0,0 +1,456 @@
...
+        it('should check the return details', async function () {
+          await testContext.addContextItem(this, 'testIdentifier', `checkReturnDetails${index}`, baseContext);
+
+          const orderReturnInfo = await returnDetailsPage.getOrderReturnInfo(page);
+          await expect(orderReturnInfo)
+            .to.contains('We have logged your return request.')
+            .and.to.contains('Your package must be returned to us within 14 days of receiving your order.')
+            .and.to.contains('The current status of your merchandise return is:')
+            .and.to.contains(test.args.status)
+            .and.to.contains('List of items to be returned:')
```
Comment (Progi1984) [inline tests/UI/campaigns/functional/FO/03_userAccount/merchandisereturns/01_consultReturnDetails.js]: Could you move strings to `returnDetailsPage` ?

File: tests/UI/campaigns/functional/FO/03_userAccount/merchandisereturns/01_consultReturnDetails.js [hunk 5]
```diff
@@ -0,0 +1,456 @@
...
+            .and.to.contains('Hummingbird printed t-shirt (Size: S - Color: White) Reference: demo_1 1');
```
Comment (Progi1984) [inline tests/UI/campaigns/functional/FO/03_userAccount/merchandisereturns/01_consultReturnDetails.js]: May be could you use demo data ?

File: tests/UI/pages/FO/myAccount/returnDetails.js [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+  constructor() {
+    super();
+
+    this.pageTitle = 'Return Details';
+    this.errorMessage = 'You must wait for confirmation before returning any merchandise.';
+
+    // Selectors
+    this.pageTitleHeader = '#main header h1.h1';
+    this.alertWarning = '#notifications .notifications-container article.alert-warning';
+    this.OrderReturnInfo = '#order-return-infos';
```
Comment (Progi1984) [inline tests/UI/pages/FO/myAccount/returnDetails.js]:
```suggestion
    this.orderReturnInfo = '#order-return-infos';
```

**Review by Progi1984 — COMMENTED** [review]
Small feedbacks :)

File: tests/UI/campaigns/functional/FO/03_userAccount/merchandisereturns/01_consultReturnDetails.js (hunk 1 above)
Comment (nesrineabdmouleh) [reply tests/UI/campaigns/functional/FO/03_userAccount/merchandisereturns/01_consultReturnDetails.js]: Done thx :+1:

File: tests/UI/pages/FO/myAccount/returnDetails.js (hunk 6 above)
Comment (nesrineabdmouleh) [reply tests/UI/pages/FO/myAccount/returnDetails.js]: Done thx :+1:

**Review by Progi1984 — APPROVED** [review]

**Review by boubkerbribri — APPROVED** [review]

**Review by NeOMakinG — APPROVED** [review]

**Pull request merged**